EpiAware/src/EpiInfModels/Renewal.jl

```diff
@@ -111,6 +116,166 @@ function (epi_model::Renewal)(recent_incidence, Rt)
+using the implicit relationship between the exponential growth rate and the reproduction
+number.
+
+```math
+\mathcal{R} \sum_{j \geq 1} g_j \exp(- r j)= 1.
+```
+
+`Renewal` are constructed by passing an `EpiData` object `data` and an
+`initialisation_prior` for the prior distribution of ``\hat{I}_0``. The default
+`initialisation_prior` is `Normal()`.
+
+## Constructor
+
+- `RenewalWithPopulation(; data, initialisation_prior, pop_size)`.
+
+## Example usage with `generate_latent_infs`
+
+`generate_latent_infs` can be used to construct a `Turing` model for the latent infections
+conditional on the sample path of a latent process. In this example, we generate a sample
+of a white noise latent process.
+
+First, we construct an `Renewal` struct with an `EpiData` object, an initialisation
+prior and a transformation function.
+
+```julia
+using Distributions, Turing, EpiAware
+gen_int = [0.2, 0.3, 0.5]
+g = exp
+
+# Create an EpiData object
+data = EpiData(gen_int, g)
+
+# Create an Renewal model
+renewal_model = RenewalWithPopulation(data = data, initialisation_prior = Normal(), pop_size = 1e6)
+```
+
+Then, we can use `generate_latent_infs` to construct a Turing model for the unobserved
+infection generation model set by the type of `direct_inf_model`.
+
+```julia
+# Construct a Turing model
+Z_t = randn(100) * 0.05
+latent_inf = generate_latent_infs(renewal_model, Z_t)
+```
+
+Now we can use the `Turing` PPL API to sample underlying parameters and generate the
+unobserved infections.
+
+```julia
+# Sample from the unobserved infections model
+
+#Sample random parameters from prior
+θ = rand(latent_inf)
+#Get unobserved infections as a generated quantities from the model
+I_t = generated_quantities(latent_inf, θ)
+```
+"
+@kwdef struct RenewalWithPopulation{S <: Sampleable} <: AbstractRenewal
+    data::EpiData
+    initialisation_prior::S = Normal()
+    pop_size::Float64 = 1e6
+end
+
+@doc """
+    function (epi_model::RenewalWithPopulation)(recent_incidence_and_available_sus, Rt)
+
+Callable on a `RenewalWithPopulation` struct for compute new incidence based on
+recent incidence, Rt and depletion of susceptibles.
+
+## Mathematical specification
+
+The new incidence is given by
+
+```math
+I_t = {S_{t-1} / N} R_t \\sum_{i=1}^{n-1} I_{t-i} g_i
+```
+
+where `I_t` is the new incidence, `R_t` is the reproduction number, `I_{t-i}` is the recent incidence
+and `g_i` is the generation interval.
+
+# Arguments
+- `recent_incidence_and_available_sus`: A tuple with an array of recent incidence
+values and the remaining susceptible/available individuals.
+- `Rt`: Reproduction number.
+
+# Returns
+- Tuple containing the updated incidence array and the new `recent_incidence_and_available_sus`
+value.
+"""
+function (epi_model::RenewalWithPopulation)(recent_incidence_and_available_sus, Rt)
+    recent_incidence, S = recent_incidence_and_available_sus
+    new_incidence = max(S / epi_model.pop_size, 0.0) * Rt *
```

Comment:
do we not want to use the correction from here: https://academic.oup.com/jrsssa/article/186/4/601/7060037

Reply:
Thats an option, there are other constructions, but yeah we don't want 

```julia
max(S / epi_model.pop_size, 0.0)
```

as the main model!

Reply:
cool shall we have this for now and upgrade using an issue?

Reply:
Yeah, the way I've got this model a (large) population actually gets depleted. I don't think the linked paper actually does depletion ($S_0$ seems fixed), which might well be fine in many contexts (e.g. infer $S_0$) and other models have history dependent feedback. The idea is that we can easily implement all of those (and indeed naturally extend to other igps).